main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -1339,13 +1339,19 @@ object Weeder {
         case e => WeededAst.Expression.Cast(e, t, f, mkSL(leftMostSourcePosition(exp), sp2))
       }
 
-    // ignoring Do for now
-    case ParsedAst.Expression.Do(sp1, op, args, sp2) =>
-      WeededAst.Expression.Hole(None, mkSL(sp1, sp2)).toSuccess
+    case ParsedAst.Expression.Do(sp1, op, args0, sp2) =>
```

Comment:
I think our use of `0` is too prolific. If the new naming strategy is `argsVal` why not just call it args then?

Reply:
Because `args` is used in the case below :)

For me the rule is: if it needs visiting, `xyz0`, if it's just passed along then just `xyz`